lib/spack/spack/url.py

```diff
@@ -56,8 +56,12 @@ def find_list_urls(url):
     GitLab     https://gitlab.\*/<repo>/<name>/tags
     BitBucket  https://bitbucket.org/<repo>/<name>/downloads/?tab=tags
     CRAN       https://\*.r-project.org/src/contrib/Archive/<name>
+    PyPI       https://pypi.org/simple/<name>/
     =========  =======================================================
 
+    Note: this function is called by `spack versions` and `spack checksum`
+    but not by `spack fetch` or `spack install`.
```

Comment:
@becker33 This is another complaint I have about our current API. The URL processing logic for `spack versions/checksum` and `spack fetch/install` is almost entirely separate. This leads to a lot of duplication, and a lot of logic that only exists in one or another.